### PR TITLE
fix(scanv4): Check if bundle contains Scanner V4 manifests

### PR DESCRIPTION
**Pull request by mclasmeier**
## Description

In the `launch_central` function in `k8sbased.sh` we try to apply manifests for Scanner V4 unconditionally. But it is not guaranteed that these manifests exist in a deployment bundle in all environments. For example, when an old roxctl (which does not have Scanner V4 support built in) is used to generate a central deployment bundle, no Scanner V4 manifests are contained in that bundle.

Therefore, what this PR does is adding an existence check for Scanner V4 manifests before trying to apply them.

@tommartensen I have assigned this to you, because I am (slightly) modifying `k8sbased.sh`. But in case you don't have the resources ATM, please ignore. Thank you.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

TODO(replace-me)  
Use this space to explain **how you validated** that **your change functions exactly how you expect it**.
Feel free to attach JSON snippets, curl commands, screenshots, etc. Apply a simple benchmark: would the information you
provided convince any reviewer or any external reader that you did enough to validate your change.

It is acceptable to assume trust and keep this section light, e.g. as a bullet-point list.

It is acceptable to skip testing in cases when CI is sufficient, or it's a markdown or code comment change only.  
It is also acceptable to skip testing for changes that are too taxing to test before merging. In such case you are
responsible for the change after it gets merged which includes reverting, fixing, etc. Make sure you validate the change
ASAP after it gets merged or explain in PR when the validation will be performed.  
Explain here why you skipped testing in case you did so.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: deploy/common/k8sbased.sh [hunk 1]
```diff
@@ -472,7 +472,13 @@ function launch_central {
           launch_service "${unzip_dir}" scanner
           if [[ "${ROX_SCANNER_V4:-}" != "false" ]]; then
             echo "Deploying ScannerV4..."
-            launch_service "${unzip_dir}" scanner-v4
+            if [[ -d "${unzip_dir}/scanner-v4" ]]; then
+              launch_service "${unzip_dir}" scanner-v4
+            else
+              echo >&2 "WARNING: Deployment bundle does not seem to contain support for Scanner V4."
+              echo >&2 "WARNING: Scanner V4 will not be deployed now."
+              echo >&2 "Possible reason for this: the roxctl in PATH does not support Scanner V4."
```
Comment (vladbologa) [inline deploy/common/k8sbased.sh:480]: I wonder if we should actually support this scenario, i.e. using an outdated roxctl. That could also lead to other subtle bugs, so arguably we could also just fail here.

**Review by vladbologa — APPROVED** [review]

File: deploy/common/k8sbased.sh (hunk 1 above)
Comment (mclasmeier) [reply deploy/common/k8sbased.sh:480]: I generally agree with these concerns. In the case at hand I think it's the best we can (and should) be doing to reduce friction during the Scanner V4 rollout, though.

**Pull request merged**